src/mixins/Perf.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Daniel Erat and Niniane Wang. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Vue from 'vue';
+import Component from 'vue-class-component';
+
+import { logDebug } from '@/firebase';
+
+@Component
+export default class Perf extends Vue {
+  constructed: DOMHighResTimeStamp;
+  stamps: Record<string, DOMHighResTimeStamp> = {};
+  loggedReady = false;
+
+  constructor() {
+    super();
+    this.constructed = window.performance.now();
+  }
+
+  // Records timestamps when the Vue and its components are mounted.
+  // This is a lifecycle hook and shouldn't be called manually.
+  mounted() {
+    this.stamp('mounted');
+    // Also report when child components have been mounted:
+    // https://vuejs.org/v2/api/#mounted
+    this.$nextTick(() => this.stamp('rendered'));
+  }
+
+  // Records a timestamp identified by |name|. Views using this mixin should
+  // call this to report significant events.
+  stamp(name: string) {
+    this.stamps[name] = window.performance.now();
+  }
+
+  // Adds a 'ready' timestamp and reports performance data. Views using this
+  // mixin should call this once when they are fully loaded.
+  logReady(code: string) {
+    if (this.loggedReady) return;
+
+    this.stamp('ready');
+
+    const payload: Record<string, any> = {};
+    for (const name of Object.keys(this.stamps)) {
+      payload[name] = Math.round(this.stamps[name] - this.constructed);
```

Comment:
Please make it clear via either comment or variable name that the payload is the milliseconds (?) elapsed since the class was constructed. When it said "performance data", that could mean memory, CPU used, etc.

Reply:
done. also renamed `stamp` to `recordEvent`